### PR TITLE
[PM-28264] Consolidate and update the UI for key connector migration/confirmation 

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-28264

## 📔 Objective

The objective is to:

-  Consolidate the `RemovePasswordComponent` into one shared HTML and angular component.
- Update the UI for key connector confirmation/migration components.

Requires the server PR https://github.com/bitwarden/server/pull/6635 for fetching the key connector confirmation details.

Team reviewer notes:

**Auth team**
Had to update a unit test because of adding a new property to `KeyConnectorDomainConfirmation`.

## 📸 Screenshots

Expand sections to see before and after.

### Confirm key connector

The UI showed to new SSO users being setup for a Key connector organization. 

<details>
<summary>
Web
</summary>

**Before**
image

**After**

after-web-confirm

</details>
<details>
<summary>
Browser extension
</summary>

**Before**
image

**After**

after-browser-confirm

</details>
<details>
<summary>
Desktop
</summary>

**Before**
image

**After**

after-desktop-confirm

</details>

### Remove master password
Converts a pre-existing master password user to a key connector user.

<details>
<summary>
Web
</summary>

**Before**
image

**After**

after-web-remove

after-web-leave-org

</details>
<details>
<summary>
Browser extension
</summary>

**Before**
image

**After**

after-browser-remove

after-browser-leave

</details>
<details>
<summary>
Desktop
</summary>

**Before**
image

**After**
after-remove-password

after-leave-org

</details>

### Tooltip 

remove password tooltip

confirm tool tip

### Edge case handling

For the key connector confirmation there is a possibility for a new client to be using an old self host server that can't provide the key connector confirmation details. If this edge case occurs the new client will still work but won't display the organization data.

edge case

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
Approving auth single test file change without reviewing rest of PR.

File: libs/key-management-ui/src/key-connector/confirm-key-connector-domain.component.ts [hunk 1]
```diff
@@ -77,4 +84,21 @@ export class ConfirmKeyConnectorDomainComponent implements OnInit {
...
+
+  private async getOrganizationName(
+    organizationSsoIdentifier: string,
+  ): Promise<string | undefined> {
+    try {
+      const details =
+        await this.keyConnectorApiService.getConfirmationDetails(organizationSsoIdentifier);
+      return details.organizationName;
+    } catch (error) {
+      // Old self hosted servers may not have this endpoint yet. On error log a warning and continue without organization name.
```
Comment (quexten) [inline libs/key-management-ui/src/key-connector/confirm-key-connector-domain.component.ts:127]: We should make a clean-up task for this as there may be potential to abuse the related vuln by the server pretending that this endpoint is not supported. (Please link it as a comment).

**Review by quexten — DISMISSED** [review]
Approving, but i've put a request for a follow-up ticket. Preferably it is linked directly here in the code.

**Comment by Thomas-Avery** [comment]
Updated all screenshots with the latest changes.

File: libs/key-management-ui/src/key-connector/confirm-key-connector-domain.component.ts (hunk 1 above)
Comment (Thomas-Avery) [reply libs/key-management-ui/src/key-connector/confirm-key-connector-domain.component.ts:127]: Added ticket and marked things to clean up.

**Review by JaredSnider-Bitwarden — APPROVED** [review]
